### PR TITLE
HDDS-11216. Replace HAUtils#buildCAX509List usages with other direct usages

### PR DESCRIPTION
**Pull request by Galsza**
## To simplify the CertificateClient the HAUtils#buildCAX509List can be refactored

There are 3 methods on DefaultCertificateClient that solely exist to serve the HAUtils method, preferably these should be removed. BuildCAList is now refactored, the places where it used the certificateClient are replaced by direct calls on the certClient.

There are a couple of technical details here: before the refactor BuildCAList went to the CertificateClient for a list of CAs, and if the CertClient didn't have that value cached it reached out to SCM. Ever since the CertificateRotation has been implemented the CertClient should be able to provide the actual current certificates even without going to the SCM. Therefore it can be supplemented by direct calls to the client.

Instead of using certificates and propagating them further, it turns out the the client can return a ClientTrustManager directly which aligns better with how the CertificateClient should look in its ideal state where only key/trustmanagers are accessible.

##
[HDDS-11216](https://issues.apache.org/jira/browse/HDDS-11216)

## How was this patch tested?

Clean CI run: https://github.com/Galsza/ozone/actions/runs/10041809622

### CODE REVIEW

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ec/reconstruction/ECReconstructionCoordinator.java [hunk 1]
```diff
@@ -118,13 +118,13 @@ public class ECReconstructionCoordinator implements Closeable {
   private final OzoneClientConfig ozoneClientConfig;
 
   public ECReconstructionCoordinator(
-      ConfigurationSource conf, CertificateClient certificateClient,
+      ConfigurationSource conf, CertificateClient certClient,
```
Comment (fapifta) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ec/reconstruction/ECReconstructionCoordinator.java:121]: Can we avoid this parameter rename please?

File: hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/HAUtils.java [hunk 2]
```diff
@@ -373,76 +372,9 @@ public static List<String> getExistingSstFiles(File db) throws IOException {
...
-  private static List<String> generateCAList(CertificateClient certClient)
-      throws IOException {
-    List<String> caCertPemList = new ArrayList<>();
-    for (X509Certificate cert : certClient.getAllRootCaCerts()) {
-      caCertPemList.add(CertificateCodec.getPEMEncodedString(cert));
-    }
-    for (X509Certificate cert : certClient.getAllCaCerts()) {
-      caCertPemList.add(CertificateCodec.getPEMEncodedString(cert));
-    }
+  private static List<X509Certificate> generateCAList(CertificateClient certClient) {
```
Comment (fapifta) [inline hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/HAUtils.java:375]: As I see this method is not used anymore, can you please remove it?

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java [hunk 3]
```diff
@@ -1610,9 +1610,9 @@ private void persistSCMCertificates() throws IOException {
     // TODO: see if we can avoid doing this during every restart.
     if (primaryScmNodeId != null && !primaryScmNodeId.equals(
         scmStorageConfig.getScmId())) {
+      getScmSecurityClientWithMaxRetry(configuration, getCurrentUser());
```
Comment (fapifta) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java:1613]: I think we don't need this line.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/storage/TestContainerCommandsEC.java
Comment (fapifta) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/storage/TestContainerCommandsEC.java:1]: I don't think we need to change indentations here, at least we should not do this as part of this PR.

**Review by fapifta — COMMENTED** [review]
Hi @Galsza sorry for the very late review, we already discussed the why's in person fortunately.
All in all the PR looks good, though I have found a few small things with it, that I would like to ask you to address before we commit this change.

Please also resolve the conflicts with the current master.

**Comment by Galsza** [comment]
@fapifta thank you for your review, please take another look. In the meanwhile there was a merge conflict with master, I think I've resolved it successfully. DNRPCLoadGenerator was conflicting

**Review by fapifta — APPROVED** [review]
Thank you for addressing the things found, and your continued efforts and patience with reviews on this one.

**Pull request merged**